tscached/shadow.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+
+def should_add_to_readahead(config, referrer, headers):
+    """ Should we add this KQuery for readahead behavior?
+        config: dict representing the top-level tscached config
+        referrer: str, from the http request
+        headers: dict, all headers from the http request
+        returns: boolean
+    """
+    if headers.get(config['shadow']['http_header_name'], None):
+        return True
+
+    for substr in config['shadow']['referrer_blacklist']:
+        if substr in referrer:
+            return False
+    return True
```

Comment:
At present, anything added to the readahead list will remain there indefinitely. This is based on the assumption that when you create a grafana dashboard you won't delete it.

We'll probably blank the whole Redis cache every so often to get rid of anything extraneous, maybe every few days or something. Remains to be seen what makes sense in production.